src/shared/hostname-setup.c

```diff
@@ -268,9 +268,10 @@ int hostname_substitute_wildcards(char *name) {
          * Note that this does not directly use the machine ID, because that's not necessarily supposed to be
          * public information to be broadcast on the network, while the hostname certainly is. */
 
-        for (char *n = name; *n; n++) {
-                if (*n != '?')
-                        continue;
+        for (char *n = name; ; n++) {
+                n = strchr(n, '?');
+                if (!n)
+                        return 0;
```

Comment:
maybe instead:

```c
for (char *n = name; *n; n = strchrnul(n + 1, '?')) 
```

Reply:
Maybe,
```c
for (char *n = strchrnul(name, '?'); *n; n = strchrnul(n + 1, '?')) 
```
I will update so.

Reply:
Hm, after reconsidering the suggestion, I prefer the current implementation.
@YHNdnzj WDYT?